cmd/task/task.go

```diff
@@ -176,18 +183,9 @@ func main() {
 		return
 	}
 
-	if list {
-		if ok := e.ListTasks(task.FilterOutInternal(), task.FilterOutNoDesc()); !ok {
-			e.Logger.Outf(logger.Yellow, "task: No tasks with description available. Try --list-all to list all tasks")
-		}
-		return
-	}
-
-	if listAll {
-		if ok := e.ListTasks(task.FilterOutInternal()); !ok {
-			e.Logger.Outf(logger.Yellow, "task: No tasks available")
-		}
-		return
+	if listOptions.ShouldListTasks() {
+		_ = e.ListTasks(listOptions)
+		return // if we are listing tasks, we don't expect a task to run
```

Comment:
Needs to exit with error if printing failed.

```suggestion
		if ok := e.ListTasks(listOptions); !ok {
			os.Exit(1)
		}
		return
```